.github/workflows/remote_index_build.yml

```diff
@@ -0,0 +1,140 @@
+name: Build and Test k-NN using Remote Index Builder
+on:
+  push:
+    branches:
+      - "*"
+      - "feature/**"
+    paths:
+      - 'build.gradle'
+      - 'settings.gradle'
+      - 'src/**'
+      - 'build-tools/**'
+      - 'buildSrc/**'
+      - 'gradle/**'
+      - 'jni/**'
+      - '.github/workflows/remote_index_build.yml'
+  pull_request:
+    branches:
+      - "*"
+      - "feature/**"
+    paths:
+      - 'build.gradle'
+      - 'settings.gradle'
+      - 'src/**'
+      - 'build-tools/**'
+      - 'buildSrc/**'
+      - 'gradle/**'
+      - 'jni/**'
+      - '.github/workflows/remote_index_build.yml'
+
+jobs:
+  Remote-Index-Build-IT-Tests:
+    strategy:
+      matrix:
+        java: [21, 23]
```

Comment:
Leave this to 21 only now as 21+23 consumed too much runners and main is still built with 21 now.

Keep 23 for other workflows for compatibility purposes.

THanks.